osu.Game/Overlays/BeatmapSet/MetadataSection.cs

```diff
@@ -116,6 +118,14 @@ private void setTextAsync(string text)
 
                         break;
 
+                    case MetadataType.Genre:
+                        loaded.AddLink(text.DehumanizeTo<SearchGenre>().GetLocalisableDescription(), LinkAction.FilterBeatmapSetGenre, text);
+                        break;
+
+                    case MetadataType.Language:
+                        loaded.AddLink(text.DehumanizeTo<SearchLanguage>().GetLocalisableDescription(), LinkAction.FilterBeatmapSetLanguage, text);
+                        break;
```

Comment:
I am not feeling very positive about this string-based flow.

* What guarantees are there that this text isn't just whatever random freeform text and that it will correctly dehumanize to a known genre/language? What happens if you search for the genre `foobar`?
* This control is going to get localisation support at some point, at which point all of this just breaks instantly.

I would feel much safer with this if it was based on actual numerical IDs that correspond to the `SearchGenre` enum - then you can filter out invalid input via `Enum.IsDefined`. These numerical IDs are already provided by web for both [genre](https://github.com/ppy/osu/blob/master/osu.Game/Beatmaps/BeatmapSetOnlineGenre.cs#L10) and [language](https://github.com/ppy/osu/blob/master/osu.Game/Beatmaps/BeatmapSetOnlineLanguage.cs#L10).

Reply:
> What guarantees are there that this text isn't just whatever random freeform text and that it will correctly dehumanize to a known genre/language? What happens if you search for the genre foobar?

This will be only used in the beatmapset overlay for now and beatmapsets can only have the "English" strings of `Search{Genre}|{Language}`. Or so I thought. The API returns the string of the language set in the website... so that breaks. The current code is also not future-proof when new genres/languages are added and will throw an exception.

> This control is going to get localisation support at some point, at which point all of this just breaks instantly.

This PR added localisation support as a bonus, see `text.DehumanizeTo<SearchLanguage>().GetLocalisableDescription()`, and works quite well despite the above problems.

---

Will change to using ids.